File: riistaandroid/src/main/java/fi/riista/mobile/models/GreySealHuntingMethod.java
package fi.riista.mobile.models;

import androidx.annotation.Nullable;

public enum GreySealHuntingMethod {

    SHOT,
    CAPTURED_ALIVE,
    SHOT_BUT_LOST;

    public static GreySealHuntingMethod fromString(@Nullable final String s) {
        return s != null ? valueOf(s) : null;
    }

    public static String toString(@Nullable final GreySealHuntingMethod type) {
        return type != null ? type.name() : null;
    }
}

File: mobile-common-lib/src/commonMain/kotlin/fi/riista/common/domain/model/GreySealHuntingMethod.kt
package fi.riista.common.domain.model

import fi.riista.common.model.BackendEnum
import fi.riista.common.model.RepresentsBackendEnum
import fi.riista.common.model.toBackendEnum
import fi.riista.common.resources.LocalizableEnum
import fi.riista.common.resources.RR

enum class GreySealHuntingMethod(
    override val rawBackendEnumValue: String,
    override val resourcesStringId: RR.string,
) : RepresentsBackendEnum, LocalizableEnum {
    SHOT("SHOT", RR.string.grey_seal_hunting_method_shot),
    CAPTURED_ALIVE("CAPTURED_ALIVE", RR.string.grey_seal_hunting_method_captured_alive),
    SHOT_BUT_LOST("SHOT_BUT_LOST", RR.string.grey_seal_hunting_method_shot_but_lost),
    ;

    // for iOS compatibility
    companion object {
        fun toBackendEnumCompat(value: String?): BackendEnum<GreySealHuntingMethod> = value.toBackendEnum()
    }
}
